MSC4300: Processing status requests & responses

Johennes opened the pull request:
[Rendered](https://github.com/gematik/matrix-spec-proposals/blob/johannes/request-response/proposals/4300-request-response.md)

Johennes committed:
MSC4300: Processing status requests & responses

Signed-off-by: Johannes Marbach <n0-0ne+github@mailbox.org>

turt2live commented on proposals/4300-request-response.md:
Implementation requirements:
* [ ] Sending client
* [ ] Receiving client (with support)
* [ ] Receiving client (without support)

Johennes committed:
Reference MSC4089

benkuly commented on proposals/4300-request-response.md:
```diff
@@ -0,0 +1,137 @@
+
+## Proposal
+
+A new content block `m.request.status` is introduced to request a processing status from other
```
Shouldn't it be `m.status.request` and `m.status.response` like the `m.key.verification.*` events?

Johennes replied:
Hm, good question. My assumption was that there might be further specialized request types in future – as in requests that ask for something else than the processing status. Therefore, I made `request` the namespace. I have no concrete examples in mind, however, and don't feel strongly about the naming either.

benkuly commented on proposals/4300-request-response.md:
```diff
@@ -0,0 +1,137 @@
+
+## Proposal
+
+A new content block `m.request.status` is introduced to request a processing status from other
```
Do we really need a request for the described use cases? Couldn't the receiver proactively just decide to send a status, when it does not support event types instead of letting the sender always request the status?

Johennes replied:
I think that could be helpful but it seems to solve a slightly different problem.

My assumption was that the sender would only add status requests on events that, for one reason or another, are important. Things such as prescriptions or sick notices where the content is not just plain text and where as the sender I need to make sure that you have actually understood and processed the event correctly. That means the majority of events would not include status requests which made adding it on the sender side appear acceptable.

Only sending the status response when an event was _not_ understood would mean that the only signal of successful processing is the absence of a response. But then you won't know if the recipient didn't respond because it understood the event or if it simply hasn't received and processed the event yet.

Johennes committed:
Use valid JSON number

Signed-off-by: Johannes Marbach <n0-0ne+github@mailbox.org>

Johennes committed:
Use vendor prefix for everything that begins with m.

Signed-off-by: Johannes Marbach <n0-0ne+github@mailbox.org>

denis-loh commented on proposals/4300-request-response.md:
````diff
@@ -0,0 +1,137 @@
+}
+```
+
+Clients that receive events containing `m.request.status` content blocks MAY respond to them with a
````
But in that case, MSC4301 and MSC4300 are directly depending on each-other, which is not immediately clear to the readers. Without the MSC4301, I still have privacy concerns as data might get transferred without acknowledgement and it's to some extend unclear how it mitigates the issue "did you understood my message?" if the other client does not implement this feature. The response event will only be sent, if the client is online, which comes with two downsides:

1. if the client is offline, the sender will not get notified, which could mean, that either the client does not support the MSC or that the recipient hasn't read the message yet.
2. if the client is online, the recipient may still not support the message or doesn't want to propagate any "online" state.

So, in neither of these cases, the sending client will ever receive the requested response, which in turn might lead to the question again "have you understood my message?"

Johennes committed:
Explain relation to MSC4301

Signed-off-by: Johannes Marbach <n0-0ne+github@mailbox.org>

ichderjens committed:
Update request-response.md with 'to_device' field

Added optional 'to_device' field for m.request.status to adress a specific receiver

Johennes committed:
Merge pull request #2 from gematik/ichderjens-patch-1

Update request-response.md with 'to_device' field

Johennes committed:
Clarify new field to_device

Signed-off-by: Johannes Marbach <n0-0ne+github@mailbox.org>

Johennes committed:
Fix typo

Signed-off-by: Johannes Marbach <n0-0ne+github@mailbox.org>

Johennes commented on proposals/4300-request-response.md:
````diff
@@ -0,0 +1,146 @@
+}
+```
+
+Clients can check whether a request has expired using `lifetime`, `origin_server_ts` and their local
````
> Can I just ask: Why specify `lifetime` and not `valid_until_ts` (or something similar)?

Mostly because `lifetime` had some precedent in the spec with [VoIP](https://spec.matrix.org/v1.18/client-server-api/#mcallinvite) (from where this proposal essentially borrows).
